perf/reader_test.go

```diff
@@ -287,6 +302,82 @@ func TestReadRecord(t *testing.T) {
 	}
 }
 
+func TestMute(t *testing.T) {
+	prog, events := mustOutputSamplesProg(t, 5)
+	defer prog.Close()
+	defer events.Close()
+
+	rd, err := NewReader(events, 4096)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rd.Close()
+
+	// Reader is already unmuted by default. It should be idempotent.
+	if err = rd.Unmute(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Write a sample. The reader should read it.
+	ret, _, err := prog.Test(make([]byte, 14))
+	if err != nil || ret != 0 {
+		t.Fatal("Can't write sample")
+	}
+	if _, err := rd.Read(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Mute. No notification should trigger.
+	if err = rd.Mute(); err != nil {
+		t.Fatal(err)
+	}
+	errChan := make(chan error)
+	go func() {
+		// Read one notification then send any errors and exit.
+		_, err := rd.Read()
+		errChan <- err
+		close(errChan)
+	}()
+	ret, _, err = prog.Test(make([]byte, 14))
+	if err == nil && ret == 0 {
+		t.Fatal("Unexpectedly rote sample while muted")
+	} // else Success
+	select {
+	case err, ok := <-errChan:
+		// Failure: Mute was unsuccessful.
+		if ok {
+			t.Fatal(err)
+		} else {
+			t.Fatal("received notification on muted reader")
+		}
+	case <-time.After(readTimeout):
+		// Success
```

Comment:
We did actually already ensure that the `prog.Test()` run failed to write the perf event, so I could see an argument that we don't need to wait around to ensure that the event never arrived here. Downside of current implementation is that it will always take 250ms if the test is passing.

Reply:
True. I think it's nice that we're testing the whole thing. Can you throw in a `t.Parallel()` at the start? 250ms isn't a huge deal I think.